modules/agar/src/main/ts/ephox/agar/api/FocusTools.ts

```diff
@@ -10,38 +10,58 @@ import * as UiControls from './UiControls';
 import * as UiFinder from './UiFinder';
 import * as Waiter from './Waiter';
 
-const cGetFocused: Chain<SugarElement<Document | ShadowRoot>, SugarElement<HTMLElement>> =
-  Chain.binder((doc) =>
-    Focus.active(doc).fold(
-      () => Result.error('Could not find active element'),
-      Result.value)
+const getFocused = (doc: SugarElement<Document | ShadowRoot>): Result<SugarElement<HTMLElement>, string> => {
+  return Focus.active(doc).fold(
+    () => Result.error('Could not find active element'),
+    Result.value
   );
+};
+
+const setFocus = <T extends Element>(container: SugarElement<Node>, selector: string): SugarElement<T> => {
+  const elem = UiFinder.findIn(container, selector).getOrDie();
```

Comment:
Does that getOrDie need a message?

Reply:
This is a result, so the message is already in the error and `getOrDie` doesn't take an error message.

Reply:
Probably.

Eventually we'll want these to be Promise failures, I think.

Reply:
These should not be promises, as they are sync. It should use `getOrDie` to throw an exception, which while then be caught be the runner and fail the test (or be caught inside the promise if it's executed inside a Promise constructor or then callback).

Reply:
Fair enough. 

I will raise, though, that using Promises to model the possibility of failure is a nice programming model. Probably not called for in TinyMCE tests, though.